controllers/eventing/cloudeventsource_controller.go

```diff
@@ -114,14 +120,25 @@ func (r *CloudEventSourceReconciler) Reconcile(ctx context.Context, req ctrl.Req
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *CloudEventSourceReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	nsFilter := util.IgnoreOtherNamespaces()
+	filter := predicate.Funcs{
+		GenericFunc: func(e event.GenericEvent) bool {
+			if e.Object.GetNamespace() == "" {
+				return true
+			}
+			return nsFilter.Generic(e)
+		},
+	}
+
 	return ctrl.NewControllerManagedBy(mgr).
+		For(&eventingv1alpha1.ClusterCloudEventSource{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
```

Comment:
Reconciliation cannot accept multiple objects. I think we should still set up multiple controllers and eliminate the duplicates through abstraction.